Hardware/Hardware/CardReader/StripeCardReader/UnderlyingError+Stripe.swift

```diff
@@ -1,9 +1,18 @@
 import StripeTerminal
+import Networking
 /// Maps Stripe SDK specific errors to domain errors:
```

Comment:
@joshheald As you mentioned in the comments, it seems like we might mixing concerns a little bit, with leads to tangle the dependency tree a little bit as well.

It seems to me like what we want is to pass up to the UI an error that is not actually thrown by the Terminal SDK, but that happens when a network request is failing. 

In one sentence: we want to pass an error to the user when a network request is failing. That request is the one that fetches the location, and is implemented by WCPayRemote.

That request fetches the location id, and what we want is to:
1. Stop the attempt to connect to the reader
2. Show a user facing error.

To me, the key is point #1. We want to stop the connection to a reader when we can not get a location identifier. That, to me, is business logic.

That's all to say that I think we should move the logic out of Hardware (which doesn't do any other networking) at least one layer up.

UnderlyigError conforms to Error, so we could do a bit of good old fashioned polymorphism. WCPayTokenProvider is the entity that encapsulates the business logic of fetching the location id. At this point, it is passing up whatever error is coming from the networking layer. 

I would explore declaring a new error alongside it (it could be an inner type to WCPayTokenProvider or it could be just a public entity declared next to it), so that WCPayTokenProvider.fetchDefaultLocationID actually throws one instance of this new error.

If everything else works as expected, this new error can be passed to CardReaderServiceError as an underlying error in when captured in StripeCardReaderService.getBluetoothConfiguration (this might or might not require adding a new case to CardReaderServiceError)

From that on, the error should be propagated up to the UI without other intervention. It might be necessary to add some logic to the UI as well to make sure that the payment intent is cancelled if connection to the reader fails.

But, in a nutshell, I would explore to:
1. Declare a new error type that would be initialized by WCPayTokenProvider when the request to get the location id fails.
2. Pass that new error in the completion block of fetchDefaultLocationID, instead of the Dotcom error that is coming from the remote service.
3. Take it from there 😄